src/init.cpp

```diff
@@ -1034,13 +1036,15 @@ bool AppInitParameterInteraction(const ArgsManager& args, bool use_syscall_sandb
         ChainstateManager::Options chainman_opts_dummy{
             .chainparams = chainparams,
             .datadir = args.GetDataDirNet(),
+            .notifications = std::make_shared<ValidationNotificationsImpl>(),
```

Comment:
Shouldn't this change be moved to the first commit?

Reply:
We also don't set other params, like `adjusted_time_callback` in the dummy opts, since they are not required to read the args into the struct. The commit where this is changed, adds it because the `BlockManager::Options` requires a reference to the notifications starting at that commit.

Reply:
FWIW, gcc emits a warning here.

Reply:
Thank you for following up, will patch.